src/Stratis.Bitcoin/Configuration/TextFileConfiguration.cs

```diff
@@ -138,13 +141,34 @@ public string[] GetAll(string key)
         /// <param name="key">Name of the argument.</param>
         /// <param name="defaultValue">Default value to return if no argument value is defined.</param>
         /// <returns>Value of the argument or a default value if no value was set.</returns>
-        public T GetOrDefault<T>(string key, T defaultValue)
+        public T GetOrDefault<T>(string key, T defaultValue, ILogger logger = null)
         {
             if (!this.args.TryGetValue($"-{key}", out List<string> values))
+            {
+                if (logger != null)
+                {
+                    if (typeof(T) == typeof(string))
+                        logger.LogDebug("('{0}':'{1}') returning default.", key, defaultValue);
```

Comment:
Please use this as a method body (fixed logging a bit)

```
public T GetOrDefault<T>(string key, T defaultValue, ILogger logger = null)
{
    logger?.LogTrace("({0}:'{1}',{2}:'{3}')", nameof(key), key, nameof(defaultValue), defaultValue);

    List<string> values;

    if (!this.args.TryGetValue($"-{key}", out values))
    {
        logger?.LogDebug("Default value '{0}' was selected for the key '{1}'.", defaultValue, key);
        logger?.LogTrace("(-)[DEFAULT]:'{0}'", defaultValue);
        return defaultValue;
    }

    try
    {
        var value = this.ConvertValue<T>(values[0]);

        logger?.LogDebug("Value '{0}' was loaded for the key '{1}'.", value, key);
        logger?.LogTrace("(-):'{0}'", value);

        return value;
    }
    catch (FormatException)
    {
        logger?.LogTrace("(-)[FORMAT_EXCEPTION]");
        throw new ConfigurationException($"Key {key} should be of type {typeof(T).Name}.");
    }
}
```

Reply:
Remove entry and exit the logs otherwise you will create a mess.

Reply:
Ok, makes sense

Reply:
Done.

Reply:
It will be a mess anyway because you are claiming calling a method in class where it is not, but at least it will be less mess without those entry and exit logs